src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1345,6 +1445,11 @@ private void AssignPatternVariables(BoundPattern pattern)
         public override BoundNode VisitBlock(BoundBlock node)
         {
             DeclareVariables(node.Locals);
+
+            // First assemble possibly-unassigned reads for all
+            // local functions
+            RecordReadsInLocalFunctions(node);
```

Comment:
I'm perfectly willing to change it in a follow-up PR. Note that this is going into a feature branch, so I can make the change before we put it in master :)